app/src/main/java/com/fstyle/structure_android/data/source/local/UserLocalDataSource.java

```diff
@@ -1,8 +1,153 @@
 package com.fstyle.structure_android.data.source.local;
 
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.support.annotation.NonNull;
+import com.fstyle.structure_android.data.model.User;
+import com.fstyle.structure_android.data.source.UserDataSource;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Created by le.quang.dao on 10/03/2017.
  */
 
-public class UserLocalDataSource {
+public class UserLocalDataSource implements UserDataSource.LocalDataSource {
+
+    private UserDbHelper mDbHelper;
+    private SQLiteDatabase mDatabase;
+
+    public UserLocalDataSource(@NonNull Context context) {
+        mDbHelper = new UserDbHelper(context);
+    }
+
+    @Override
+    public void openTransaction() {
+        mDatabase = mDbHelper.getWritableDatabase();
+        mDatabase.beginTransaction();
+    }
+
+    @Override
+    public void closeTransaction() {
+        mDatabase.close();
+    }
+
+    @Override
+    public void openReadTransaction() {
+        mDbHelper.getReadableDatabase();
+    }
+
+    @Override
+    public void insertUser(@NonNull User user) {
+        if (mDatabase == null || !mDatabase.isOpen()) {
+            throw new RuntimeException("Need call openTransaction");
```

Comment:
WARNING: [AvoidDuplicateLiterals] The String literal "Need call openTransaction" appears 4 times in this file; the first occurrence is on line 45
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals